### PR TITLE
add per-module optimization level

### PR DESCRIPTION
**Pull request by JeffBezanson**
Time for `display(plot(rand(10)))` on Plots master, baseline:
```
 25.938760 seconds (18.25 M allocations: 899.022 MiB, 1.74% gc time)
```
On this branch, adding `Base.@optlevel 0` to Plots.jl:
```
 20.341163 seconds (18.26 M allocations: 899.217 MiB, 2.30% gc time)
```

### CODE REVIEW

**Comment by vtjnash** [comment]
It might be better to map this directly to LLVM's `optnone` function attribute on the containing function itself, so that we don't have to inject the extra global state into the JIT. But otherwise, that's an impressive speedup!

**Comment by JeffBezanson** [comment]
Ready for review.

Now that we have #35157, this should be able to work without affecting other functions that happen to be in the same llvm module.

**Review by vtjnash — COMMENTED** [review]
Do you have new numbers for this?

File: src/jitlayers.cpp [hunk 1]
```diff
@@ -436,11 +436,68 @@ void JuliaOJIT::DebugObjectRegistrar::operator()(RTDyldObjHandleT H,
...
     JL_TIMING(LLVM_OPT);
-    jit.PM.run(M);
+    int optlevel;
+    if (jl_generating_output()) {
+        optlevel = 0;
+    }
+    else {
+        optlevel = jl_options.opt_level;
+        for (Module::iterator fs = M.begin(); fs != M.end(); ) {
+            Function &F = *fs;
```
[the remark this answers is not in the document]
Comment (JeffBezanson) [reply src/jitlayers.cpp:473]: Agreed, that sounds better.

**Comment by JeffBezanson** [comment]
With this, setting `@optlevel 1` in Plots.jl takes the time from 18.5 sec to 13.5 sec.

**Comment by JeffBezanson** [comment]
Ok I think this is ready to go.

File: src/jitlayers.cpp [hunk 2]
```diff
@@ -436,11 +436,63 @@ void JuliaOJIT::DebugObjectRegistrar::operator()(RTDyldObjHandleT H,
...
 {
     JL_TIMING(LLVM_OPT);
-    jit.PM.run(M);
+    int optlevel;
+    if (jl_generating_output()) {
+        optlevel = 0;
+    }
+    else {
+        optlevel = jl_options.opt_level;
+        for (Module::iterator fs = M.begin(); fs != M.end(); ) {
```
Comment (vtjnash) [inline src/jitlayers.cpp:470]:
```suggestion
        for (auto &F : M.functions()) {
```

**Review by vtjnash — APPROVED** [review]

**Comment by chriselrod** [comment]
I think this commit caused https://github.com/JuliaLang/julia/issues/35393#issuecomment-610799156

**Pull request merged**